messageapi/src/main/java/fi/hel/verkkokauppa/message/service/MessageService.java

```diff
@@ -0,0 +1,52 @@
+package fi.hel.verkkokauppa.message.service;
+
+import fi.hel.verkkokauppa.common.util.UUIDGenerator;
+import fi.hel.verkkokauppa.message.dto.MessageDto;
+import fi.hel.verkkokauppa.message.enums.MessageTypes;
+import fi.hel.verkkokauppa.message.model.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.MailException;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Component;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+@Component
+public class MessageService {
+
+    private final Logger log = LoggerFactory.getLogger(MessageService.class);
+
+    @Autowired
+    private JavaMailSender javaMailSender;
+
+    public Message createSendableEmailMessage(MessageDto messageDto) {
+        return new Message(
+                UUIDGenerator.generateType3UUIDString(MessageTypes.EMAIL.toString(), messageDto.getOrderId()),
```

Comment:
Does this generate new ID everytime with same parameters?

Reply:
This needs to be changed if multiple messages regarding same order are ever sent. (Type 3 generates the same id with same parameters.) There was initially a random generated id method available.

Reply:
- [ ] Change UUID generator to use UUIDV4